packages/frontend/src/lib/disk-image/DiskImageDetailsVirtualMachine.svelte

```diff
@@ -0,0 +1,309 @@
+  logsTerminal.onKey((e: any) => {
+    e = e.key;
+    if (socket !== undefined) {
+      const encoder = new TextEncoder();
+      const binaryData = encoder.encode(e);
+      socket.send(binaryData.buffer);
+    }
+  });
+
+  // Allow copy / paste support with Ctrl+V
+  logsTerminal.attachCustomKeyEventHandler((arg: KeyboardEvent) => {
+    // Allow copy and paste from the ctrl key (windows and linux)
+    // and "meta" key (macOS). Sometimes on macOS the ctrl key is used, so we safely assume that either meta or ctrl is meant
+    // for paste.
+    if ((arg.ctrlKey || arg.metaKey) && arg.code === 'KeyV' && arg.type === 'keydown') {
+      bootcClient.readFromClipboard().then(clipboardText => {
+        // Send to socket
+        if (socket !== undefined) {
+          const encoder = new TextEncoder();
+          const binaryData = encoder.encode(clipboardText);
+          socket.send(binaryData.buffer);
+        }
+      });
+    }
+    return true;
+  });
+}
+
+// Launch the VM with the folder location and architecture required.
+async function launchVM(build: BootcBuildInfo): Promise<void> {
+  launchInProgress = true;
+
+  // This is launched IN THE BACKGROUND. We do not wait for the VM to boot before showing the terminal.
+  // we instead are notified by subscribing to Messages.MSG_VM_LAUNCH_ERROR messages from RPC
+  bootcClient.launchVM(build);
+
+  // Initialize the terminal so it awaits the websocket connection.
+  await initTerminal();
+
+  // To avoid a blank terminal wait until terminal has logs and and then show it
+  // logs.terminal.buffer.normal will contain the "ascii cursor" with a value of 1 until there is more logs.
+  // we wait until buffer.normal.length is more than 1.
+  while (logsTerminal.buffer.normal.length < 1) {
+    await new Promise(resolve => setTimeout(resolve, 500));
+  }
+
+  // Reset the terminal if we are booting an arm64 VM on macOS
+  // see function for more information why this is necessary.
+  if (build.arch === 'arm64' && (await bootcClient.isMac())) {
+    await resetTerminalTheme();
+  }
+}
+
+// This is a temporary workaround for a terminal theme issue that ONLY happens when arm64 VMs are launched
+// on macOS. The reasoning is that it is for some reason using the default terminal theme despite -nographic being set.
+// this could possibly be because we use host CPU and memory as well as the 'virt' module for the VM to be working / launched
+// correctly in accelerated mode.
+//
+// To work "around" this, we reset the theme 10 lines after the VM is booted in order to clear any colourization ASCII characters that may
+// be present in the terminal.
+async function resetTerminalTheme(): Promise<void> {
+  // Wait until we have more than 100 lines in the buffer before resetting the terminal
+  while (logsTerminal.buffer.normal.length < 100) {
+    await new Promise(resolve => setTimeout(resolve, 500));
+  }
+
+  // Reset the terminal
+  logsTerminal.reset();
+}
```

Comment:
it doesn't reset the theme, it reset the terminal

Reply:
That's intentional as it will clear any ASCII background characters in the log output. Tried many other alternative ways (including "setting the theme again") and nothing works.

It's a bug with websockets mode and using the host hardware for mac support and luckily a very minor edge case.